### PR TITLE
library bulk updates

### LINKED ISSUE #150: New geth version released: Mawinor (v1.12.2)
### What was wrong?<br>

* Support is needed for newly released geth version: `v1.12.2`<br>

### How can it be fixed?<br>
* Add support for geth version `v1.12.2`

### LINKED ISSUE #155: New geth version released: Therum (v1.13.0)
### What was wrong?<br>

* Support is needed for newly released geth version: `v1.13.0`<br>

### How can it be fixed?<br>
* Add support for geth version `v1.13.0`

### LINKED ISSUE #156: New geth version released: Zakros (v1.13.1)
### What was wrong?<br>

* Support is needed for newly released geth version: `v1.13.1`<br>

### How can it be fixed?<br>
* Add support for geth version `v1.13.1`

### LINKED ISSUE #157: New geth version released: Phaistos (v1.13.2)
### What was wrong?<br>

* Support is needed for newly released geth version: `v1.13.2`<br>

### How can it be fixed?<br>
* Add support for geth version `v1.13.2`

### LINKED ISSUE #158: New geth version released: Armeni (v1.13.3)
### What was wrong?<br>

* Support is needed for newly released geth version: `v1.13.3`<br>

### How can it be fixed?<br>
* Add support for geth version `v1.13.3`

### LINKED ISSUE #159: New geth version released: Archanes (v1.13.4)
### What was wrong?<br>

* Support is needed for newly released geth version: `v1.13.4`<br>

### How can it be fixed?<br>
* Add support for geth version `v1.13.4`

### PR DESCRIPTION
**Pull request by fselmo**
### What was wrong?

We need to catch up on geth support. Let's lump it all in here and address any updates in this PR.

### How was it fixed?

- Add support for geth `v1.12.2` to `v1.13.4`:
  - closes #150 
  - closes #155 
  - closes #156 
  - closes #157 
  - closes #158 
  - closes #159 

- Drop support for geth `v1.9` series and `v1.10` series. Shanghai network upgrade, along with support for withdrawals, was introduced in `v1.11.0`. This feels like a good place to draw the line.
- Drop support for Python 3.7 - EOL was reached in June
- Use golang `v1.21.3` for CircleCI builds

### Todo:
- [x] Clean up commit history
- [x] Add or update documentation related to these changes
- [x] Add entry to the [release notes](https://github.com/ethereum/py-geth/blob/master/newsfragments/README.md)

### CODE REVIEW

**Review by reedsa — APPROVED** [review]
No concerns here, lgtm!

File: README.md
Comment (pacrob) [inline README.md:1]: Still says back to v1.9.14 can be installed on line125.

**Review by pacrob — APPROVED** [review]
Just the version update in README, otherwise lgtm!

File: README.md
Comment (fselmo) [reply README.md:1]: eek good catch

**Pull request merged**